L1Trigger/L1THGCal/plugins/veryfrontend/HGCalVFEProcessorSums.cc

```diff
@@ -83,8 +83,8 @@ void HGCalVFEProcessorSums::run(const HGCalDigiCollection& digiColl,
   for (const auto& [tc_id, tc_value] : tc_payload) {
     if (tc_value > 0) {
       const auto& [tc_compressed_code, tc_compressed_value] = tc_compressed_payload[tc_id];
-      l1t::HGCalTriggerCell triggerCell(reco::LeafCandidate::LorentzVector(), tc_compressed_value, 0, 0, 0, tc_id);
-      triggerCell.setCompressedCharge(tc_compressed_code);
+      l1t::HGCalTriggerCell triggerCell(reco::LeafCandidate::LorentzVector(), static_cast<int>(tc_compressed_value), 0, 0, 0, tc_id);
```

Comment:
Might be good to check that the value fits into an `int`

Reply:
Yes, just I am not yet sure what to do if the value does not fit into an `int`, so I just truncate the bits for now...

Reply:
@jbsauvan Or do you think it is wise to generate an exception if the value does not fit?

Reply:
Maybe not an exception, but at least a warning message.

Reply:
Good idea. I have just implemented the warnings.